src/ext/java/org/opentripplanner/ext/legacygraphqlapi/datafetchers/LegacyGraphQLstepImpl.java

```diff
@@ -28,6 +95,41 @@ public DataFetcher<Double> lon() {
     return environment -> getSource(environment).getStartLocation().longitude();
   }
 
+  @Override
+  public DataFetcher<LegacyGraphQLRelativeDirection> relativeDirection() {
+    return environment ->
+      switch (getSource(environment).getRelativeDirection()) {
+        case DEPART -> LegacyGraphQLRelativeDirection.DEPART;
+        case HARD_LEFT -> LegacyGraphQLRelativeDirection.HARD_LEFT;
+        case LEFT -> LegacyGraphQLRelativeDirection.LEFT;
+        case SLIGHTLY_LEFT -> LegacyGraphQLRelativeDirection.SLIGHTLY_LEFT;
+        case CONTINUE -> LegacyGraphQLRelativeDirection.CONTINUE;
+        case SLIGHTLY_RIGHT -> LegacyGraphQLRelativeDirection.SLIGHTLY_RIGHT;
+        case RIGHT -> LegacyGraphQLRelativeDirection.RIGHT;
+        case HARD_RIGHT -> LegacyGraphQLRelativeDirection.HARD_RIGHT;
+        case CIRCLE_CLOCKWISE -> LegacyGraphQLRelativeDirection.CIRCLE_CLOCKWISE;
+        case CIRCLE_COUNTERCLOCKWISE -> LegacyGraphQLRelativeDirection.CIRCLE_COUNTERCLOCKWISE;
+        case ELEVATOR -> LegacyGraphQLRelativeDirection.ELEVATOR;
+        case UTURN_LEFT -> LegacyGraphQLRelativeDirection.UTURN_LEFT;
+        case UTURN_RIGHT -> LegacyGraphQLRelativeDirection.UTURN_RIGHT;
+      };
+  }
+
+  @Override
+  public DataFetcher<Boolean> stayOn() {
+    return environment -> getSource(environment).getStayOn();
+  }
+
+  @Override
+  public DataFetcher<String> streetName() {
+    return environment -> getSource(environment).getStreetName().toString(environment.getLocale());
```

Comment:
You could use `org.opentripplanner.ext.legacygraphqlapi.LegacyGraphQLUtils#getTranslation` if we want to enable fetching different translations here.